src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -918,6 +1026,37 @@ subroutine read_fv3_netcdf_guess(fv3filenamegin)
           write(6,*)'fv3lam_io_dynmetvars2d_nouv is ',(trim(fv3lam_io_dynmetvars2d_nouv(i)),i=1,ntracerio3d)
       endif      
 
+    if (laeroana_fv3cmaq) then
+      jtracer = 0
+!name_chemvars3d chemguess from anainfo
+      do i=1,size(name_chemvars3d)
+        vartem=trim(name_chemvars3d(i))
+           if (ifindstrloc(aeronames_cmaq_fv3,trim(vartem)) > 0) then
+             jtracer=jtracer+1
+             fv3lam_io_tracerchemvars3d_nouv(jtracer)=trim(vartem)
```

Comment:
A question: the chem. vars are also in the tracer file.  If the IO of chem vars are delt along with other tracer files,
namely, combining fv3lam_io_tracerchemvars3d_nouv ( and corressponding IO bundle ) and  fv3lam_io_tracervars3d_nouv,this will cause a more efficient/scalable parallel IO. 
 Will this add any significant complicities to the coding?

Reply:
This is a good question/suggestion. I thought about this when I updated my code. If you think the combination will improve the efficiency of parallel IO, I may reconsider it. Thanks.

Reply:
The current parallelization is done over number of variable number multiplied by the number of vertical levels.  Assume, vertical levels are 60 levels, in the original tracers, there are two tracer variables , and , 7 chem.  variables.  In the current scheme, first,  2*60  fields would be distributed to various MPI processes and their IO (and conversion between native and analysis grids) would be done in parallel; second, after the first step, the similar would be done chem. variable, now, it is 7*60 fields would be deal with in parallel.  If they are done together, there would be only one step:  (2+7)*60 fields would be dealt with in parallel.  In this case, if the number of mpi process is more than 120, then, in the first step of the current scheme,  some MPI process would do nothing and namely the scalibility is not so well.  Hope my explanation will help!